### PR TITLE
[PYTHON] Only generate protos in development

### LINKED ISSUE #333: 3.0.0-alpha-2 won't pip install
https://pypi.python.org/pypi/protobuf/3.0.0-alpha-2

For example:

```
$ virtualenv venv
New python executable in venv/bin/python
Installing setuptools, pip...done.
$ source venv/bin/activate
(venv) $ pip install protobuf==3.0.0-alpha-2
Collecting protobuf==3.0.0-alpha-2
  Using cached protobuf-3.0.0-alpha-2.tar.gz
Requirement already satisfied (use --upgrade to upgrade): setuptools in ./venv/lib/python2.7/site-packages (from protobuf==3.0.0-alpha-2)
Installing collected packages: protobuf
  Running setup.py install for protobuf
    Complete output from command /path/venv/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip-build-oncqNy/protobuf/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-E2WkTi-record/install-record.txt --single-version-externally-managed --compile --install-headers /path/venv/include/site/python2.7/protobuf:
    /path/venv/local/lib/python2.7/site-packages/setuptools/dist.py:282: UserWarning: Normalizing '3.0.0-alpha-2' to '3.0.0a2'
      normalized_version,
    running install
    running build
    running build_py
    Can't find required file: ../src/google/protobuf/unittest.proto
    Generating google/protobuf/unittest_pb2.py...

    ----------------------------------------
    Command "/path/venv/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip-build-oncqNy/protobuf/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-E2WkTi-record/install-record.txt --single-version-externally-managed --compile --install-headers /path/venv/include/site/python2.7/protobuf" failed with error code 255 in /tmp/pip-build-oncqNy/protobuf
```

Discovered in https://travis-ci.org/GoogleCloudPlatform/gcloud-python/builds/60881924
/cc @tseaver

### PR DESCRIPTION
**Pull request by tamird**
Fixes #333.

@haberman @tseaver

### CODE REVIEW

**Comment by tseaver** [comment]
I don't believe removing MANIFEST.in helps:  that file governs what files are included when creating the "sdist", but is not in play during installation.  The `build_py` step is what fails on install.

**Comment by tamird** [comment]
I tested it locally (and added it to CI) so it seems to work. It's another matter that the cpp_implementation tests are failing now :(

**Comment by tseaver** [comment]
@tamird have you tested it by running `python setup.py sdist` and then trying to install the generated sdist in a fresh virtualenv? (not anywhere inside the main protobuf repo)?

**Comment by tseaver** [comment]
#360 out-of-lines the proto-generation stuff from "normal" installation, adding a new `setup.py generate_protos` command for the maintainers to use.

**Comment by tamird** [comment]
Hey, this is green (and it tests `pip install` in a fresh virtualenv). How about it @haberman ?

**Comment by haberman** [comment]
I love the code to test pip in a virtualenv.

I would rather not remove MANIFEST.in, because it pollutes our distribution with stuff that doesn't belong there.

Instead, to fix the build, could we simply add this at the beginning of GenerateUnittestProtos():

```python
# Unittest protos are only needed for development.
if not os.path.exists("../.git"):
  return
```

File: travis.sh [hunk 1]
```diff
@@ -102,10 +104,13 @@ build_python_cpp() {
   ./autogen.sh
   ./configure
   make -j2
-  export LD_LIBRARY_PATH=../src/.libs
+  export   LD_LIBRARY_PATH=../src/.libs
+  export DYLD_LIBRARY_PATH=../src/.libs
```
Comment (haberman) [inline travis.sh]: Please add a comment like "# For OS X"

**Comment by tamird** [comment]
Done.

File: travis.sh (hunk 1 above)
Comment (tamird) [inline travis.sh]: done

**Comment by haberman** [comment]
Thanks a lot! Merging.

**Pull request merged**